src/current/v24.3/ldap-authentication.md

```diff
@@ -0,0 +1,149 @@
+
+Set the custom CA certificate:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap_authentication.domain_ca = '<PEM_ENCODED_CA_CERT>';
+~~~
+
+Configure a client certificate for mTLS if required:
+
+~~~ sql
+SET CLUSTER SETTING server.ldap_authentication.client.tls_certificate = '<PEM_ENCODED_CERT>';
+SET CLUSTER SETTING server.ldap_authentication.client.tls_key = '<PEM_ENCODED_KEY>';
+~~~
+
+### Step 4: Sync Database Users
+
+Before LDAP authentication can be used for a user, the username must be created directly in CockroachDB. You will, therefore, need to establish an automated method for keeping users in sync with the directory server, creating and dropping them as needed.
+
+For Azure AD deployments, the username typically corresponds to the `sAMAccountName` field from AD's person object, which is specified in the HBA configuration using `ldapsearchattribute=sAMAccountName`.
+
+{{site.data.alerts.callout_info}}
+SQL usernames must comply with CockroachDB's [username requirements](https://www.cockroachlabs.com/docs/v24.2/create-user#user-names). Ensure that the values in the LDAP `sAMAccountName` field (or other mapped attribute) meet these requirements.
+{{site.data.alerts.end}}
+
+To create a single user:
+
+~~~ sql
+CREATE ROLE username LOGIN;
+~~~
+
+To create users in bulk:
+
+1. Export usernames from the directory server.
+1. Produce a `.sql` file with a [`CREATE ROLE`](https://www.cockroachlabs.com/docs/v24.2/create-role) statement per user, each on a separate line.
+
+    ~~~ sql
+    CREATE ROLE username1 LOGIN;
+    CREATE ROLE username2 LOGIN;
+    CREATE ROLE username3 LOGIN;
+    ~~~
+
+    If you are not also enabling LDAP Authorization to manage roles and privileges, you can also include one or more `GRANT` lines for each user. For example, `GRANT developer TO username1` or `GRANT SELECT ON DATABASE orders TO username2;`.
+
+1. Run the SQL file to create the users.
+
+    ~~~ sql
+    cockroach sql --file=create_users.sql --host=<servername> --port=<port> --user=<user> --database=<db> --certs-dir=path/to/certs
+    ~~~
+
+You can similarly [`DROP`](https://www.cockroachlabs.com/docs/stable/drop-role) users as they are removed from the directory, and update roles and groups as needed, if not using LDAP authorization.
+
+### Connecting to CockroachDB
+
+#### SQL shell connection with LDAP authentication
+
+After LDAP authentication is configured, users can connect using their LDAP credentials:
+
+~~~ bash
+# Method 1: Password in connection string
+cockroach sql --url "postgresql://username:ldap_password@host:26257" --certs-dir=certs
+
+# Method 2: Password in environment variable
+export PGPASSWORD='ldap_password'
+cockroach sql --url "postgresql://username@host:26257" --certs-dir=certs
+~~~
```

Comment:
I would not bury this detail in comments. I'd pull it out into para text:

You can store the LDAP password directly in the connection string, or you can set it in the `PGPASSWORD` environment variable before connecting.

Are there security concerns about either approach here? The password will be in your shell history if you use either of these options. It's a more valuable password than just a cluster one, because it potentially gets you into the Azure tenant itself.

Reply:
Passing passwords/secrets through env vars is a pretty common practice.